src/annotator/guest.js

```diff
@@ -293,25 +291,24 @@ export default class Guest {
     });
 
     this.crossframe.on('scrollToAnnotation', tag => {
-      for (let anchor of this.anchors) {
-        if (anchor.highlights) {
-          if (anchor.annotation.$tag === tag) {
-            const range = resolveAnchor(anchor);
-            if (!range) {
-              continue;
-            }
-
-            const event = new CustomEvent('scrolltorange', {
-              bubbles: true,
-              cancelable: true,
-              detail: range,
-            });
-            const defaultNotPrevented = this.element.dispatchEvent(event);
-            if (defaultNotPrevented) {
-              scrollIntoView(anchor.highlights[0]);
-            }
-          }
-        }
+      const anchor = this.anchors.find(a => a.annotation.$tag === tag);
```

Comment:
The previous implementation was a loop (looking for and acting on anchors with matching `$tag`s) while the new variant only operates on the first anchor whose `$tag` matches. My sense is that the previous version had an implicit assumption that 0 or 1 anchor `$tag`s would match, but didn't explicitly break out of the loop after finding that first match. Correct?

My further sense is that the new variant is much more explicit that we only want to scroll to one thing, and that thing should be the _first_ anchor with a matching `$tag` (conceptually, there should only be one anchor with a `$tag` that matches, of course). So, bravo.

Reply:
> My sense is that the previous version had an implicit assumption that 0 or 1 anchor $tags would match, but didn't explicitly break out of the loop after finding that first match. Correct?

Yes, exactly. Also since it is only possible to scroll the document to a single position at a time, it makes sense to pick the first anchor and ignore all the others.